.github/workflows/helpers/formula-common.sh

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+empty_sha="dummySHA256sum"
+
+pascal_case() {                       # gardenctl-v2 → GardenctlV2
+  IFS='-' read -ra parts <<<"$1"; local out=""; for p in "${parts[@]}"; do
+    out+=${p^}; done; echo "$out"
+}
+
+render_formula() {
+  local component=$1 tag=$2 desc=$3 extra_dep=$4 extra_install=$5 caveats=$6 extra_tests=$7
+  local version=${tag#v} org=${GITHUB_REPOSITORY_OWNER} class_name
+  class_name=$(pascal_case "$component")
+
+  cat > "${component}.rb" <<EOF
+# typed: true
+# frozen_string_literal: true
+
+class ${class_name} < Formula
+  desc      "${desc}"
+  homepage  "https://gardener.cloud"
+  version   "${version}"
+${extra_dep}
+
+  if OS.mac?
+    if Hardware::CPU.arm?
+      url "https://github.com/${org}/${component}/releases/download/${tag}/${component//-/_}_darwin_arm64"
+      sha256 "${darwin_sha_arm64}"
+    else
+      url "https://github.com/${org}/${component}/releases/download/${tag}/${component//-/_}_darwin_amd64"
+      sha256 "${darwin_sha_amd64}"
+    end
+  elsif OS.linux?
+    if Hardware::CPU.arm?
+      url "https://github.com/${org}/${component}/releases/download/${tag}/${component//-/_}_linux_arm64"
+      sha256 "${linux_sha_arm64}"
+    else
+      url "https://github.com/${org}/${component}/releases/download/${tag}/${component//-/_}_linux_amd64"
+      sha256 "${linux_sha_amd64}"
+      depends_on arch: :x86_64
+    end
```

Comment:
The common script assumes binaries use _ before the platform (e.g., gardenctl_v2_darwin_arm64 for gardenctl-v2).
But for diki, the old script used - (e.g., diki-darwin-arm64). -> The generated URL will be incorrect (e.g., diki_darwin_arm64 instead).
Proper url example for diki: https://github.com/gardener/diki/releases/download/v0.18.0/diki-darwin-amd64

Add a parameter for the separator (e.g., _ or -) in render_formula, or get diki component to change their asset naming convention :)

Reply:
I reverted to the deicated formula files. I think it makes more sense at this point in time